pennylane/circuit_drawer.py

```diff
@@ -0,0 +1,916 @@
+    def to_superscript(num):
+        """Convert the given number to a superscripted string."""
+        pass
+
+
+class UnicodeCharSet(CharSet):
+    """Charset for CircuitDrawing made of Unicode characters."""
+
+    # pylint: disable=too-few-public-methods
+
+    WIRE = "─"
+    MEASUREMENT = "┤"
+    TOP_MULTI_LINE_GATE_CONNECTOR = "╭"
+    MIDDLE_MULTI_LINE_GATE_CONNECTOR = "├"
+    BOTTOM_MULTI_LINE_GATE_CONNECTOR = "╰"
+    EMPTY_MULTI_LINE_GATE_CONNECTOR = "│"
+    CONTROL = "C"
+    LANGLE = "⟨"
+    RANGLE = "⟩"
+    VERTICAL_LINE = "│"
+    CROSSED_LINES = "╳"
+    PIPE = "|"
+    OTIMES = "⊗"
+
+    @staticmethod
+    def to_superscript(num):
+        """Convert the given number to a superscripted string."""
+        ret = str(num)
+        for old, new in {
+            "0": "⁰",
+            "1": "¹",
+            "2": "²",
+            "3": "³",
+            "4": "⁴",
+            "5": "⁵",
+            "6": "⁶",
+            "7": "⁷",
+            "8": "⁸",
+            "9": "⁹",
+        }.items():
+            ret = ret.replace(old, new)
+
+        return ret
+
+    @staticmethod
+    def to_subscript(num):
+        """Convert the given number to a subscripted string."""
+        ret = str(num)
+        for old, new in {
+            "0": "₀",
+            "1": "₁",
+            "2": "₂",
+            "3": "₃",
+            "4": "₅",
+            "5": "⁵",
+            "6": "₆",
+            "7": "₇",
+            "8": "₈",
+            "9": "₉",
+        }.items():
+            ret = ret.replace(old, new)
+
+        return ret
+
+
+class AsciiCharSet(CharSet):
+    """Charset for CircuitDrawing made of Unicode characters."""
+
+    # pylint: disable=too-few-public-methods
+
+    WIRE = "-"
+    MEASUREMENT = "|"
+    TOP_MULTI_LINE_GATE_CONNECTOR = "+"
+    MIDDLE_MULTI_LINE_GATE_CONNECTOR = "+"
+    BOTTOM_MULTI_LINE_GATE_CONNECTOR = "+"
+    EMPTY_MULTI_LINE_GATE_CONNECTOR = "|"
+    CONTROL = "C"
+    LANGLE = "<"
+    RANGLE = ">"
+    VERTICAL_LINE = "|"
+    CROSSED_LINES = "X"
+    PIPE = "|"
+    OTIMES = "@"
+
+    @staticmethod
+    def to_superscript(num):
+        """Convert the given number to a superscripted string."""
+        return "^" + str(num)
+
+    @staticmethod
+    def to_subscript(num):
+        """Convert the given number to a subscripted string."""
+        return "_" + str(num)
+
+
+Charsets = {
+    "unicode": UnicodeCharSet,
+    "ascii": AsciiCharSet,
+}
+"""Dictionary with all available CharSets."""
+
+
+class RepresentationResolver:
+    """Resolves the string representation of PennyLane objects.
+
+    Args:
+        charset (CharSet, optional): The CharSet to be used for representation resolution. Defaults to UnicodeCharSet.
+        show_variable_names (bool, optional): Show variable names instead of variable values. Defaults to False.
+    """
+
+    # Symbol for uncontrolled wires
+    resolution_dict = {
+        "PauliX": "X",
+        "CNOT": "X",
+        "Toffoli": "X",
```

Comment:
This might be worth considering:
https://www.compart.com/en/unicode/U+2295

Reply:
The problem here is that those special unicode characters usually do not exist in the monospace fonts, which makes the display of the circuit _really ugly_. At the end of the circuit it doesn't matter, inside it does:

```
 0: ──RX(a)───────────────────────╭C────RX(angles[0])───────────────────────────────╭C─────╭C─────╭C──╭C──────────╭C──╭SWAP───╭SWAP───┤ ⟨Y⟩
 1: ──RX(b)────────Z──────────────╰⊕───╭RY(b)──────────RX(4*angles[1])──╭RY(0.359)──├⊕──Z──│───Z──╰Z──│───╭⊕──╭C──│───│───────├SWAP───┤ Var[H]
 2: ──RY(1.889*c)──RX(angles[2])───U0──│────────────────────────────────│───────────╰C─────╰⊕─────────╰Z──╰C──│───╰⊕──╰SWAP───│───────┤ Sample[X]
 3: ───────────────────────────────────╰C──────────────RZ(b)────────────╰C────────────────────────────────────╰⊕───────RZ(b)──│──────╭┤ ⟨H0⟩
 4: ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╰C─────╰┤ ⟨H0⟩
```

Reply:
True that!